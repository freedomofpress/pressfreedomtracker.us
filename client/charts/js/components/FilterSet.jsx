import React, { useContext } from 'react'
import PropTypes from 'prop-types'
import TagFilter from './TagFilter'
import { removeElement } from '../lib/utilities'
import {
	SET_PARAMETER,
} from '../lib/actionTypes'
import { FiltersDispatch } from '../lib/context'


function isDateValid(date) {
	return !Number.isNaN(new Date(date).getYear())
}


function DateFilter({
	name,
	label,
	value,
}) {
	const updateFilters = useContext(FiltersDispatch);
	let lowerId = `id_${name}_after`
	let upperId = `id_${name}_before`
	let lowerValue = value.min?.toISOString()?.split('T')[0]
	let upperValue = value.max?.toISOString()?.split('T')[0]
	return (
		<>
			<div className={"filters__field-row"}>
				<div className="filters__field-label">
					<label htmlFor={lowerId}>{label} after</label>
				</div>
				<input
					type="date"
					value={lowerValue ?? ""}
					name={name}
					id={lowerId}
					onChange={(event) => {
						let newMinDate = event.target.value
						newMinDate = isDateValid(newMinDate) ? new Date(newMinDate) : null
						// maybe change this to SET_MIN_PARAMETER if
						// it is easy to do.
						updateFilters({
							type: SET_PARAMETER,
							payload: {
								filterName: name,
								value: { min: newMinDate, max: upperValue }
							},
						})
					}}
				/>
			</div>
			<div className={"filters__field-row"}>
				<div className="filters__field-label">
					<label htmlFor={upperId}>{label} before</label>
				</div>
				<input
					type="date"
					name={name}
					value={upperValue ?? ""}
					id={upperId}
					onChange={(event) => {
						let newMaxDate = event.target.value
						newMaxDate = isDateValid(newMaxDate) ? new Date(newMaxDate) : null
						updateFilters({
							type: SET_PARAMETER,
							payload: {
								filterName: name,
								value: { min: lowerValue, max: newMaxDate }
							},
						})
					}}
				/>
			</div>
		</>
	)
}

function TextFilter({
	name,
	label,
	value,
	handleFilterChange,
}) {
	let id = `id_${name}`
	return (
		<div className={"filters__field-row"}>
			<div className={"filters__field-label"}>
				<label htmlFor={id}>{label}</label>
			</div>

			<input
				value={value}
				type="text"
				name={name}
				className="text-field--single"
				autoComplete="off"
				onChange={handleFilterChange}
				id={id}
			/>
		</div>
	)
}

TextFilter.propTypes = {
	name: PropTypes.string.isRequired,
	label: PropTypes.string.isRequired,
	value: PropTypes.string,
	handleFilterChange: PropTypes.func.isRequired,
}

function SelectFilter({
	name,
	label,
	value,
	choices,
	handleFilterChange,
}) {
	let id = `id_${name}`
	return (
		<div className={"filters__field-row"}>
			<div className="filters__field-label">
				<label htmlFor={id}>{label}</label>
			</div>
			<select
				id={id}
				name={name}
				value={value}
				onChange={handleFilterChange}
			>
				<option value="">------</option>
				{choices.map(([value, choice]) => (
					<option
						key={value}
						value={value}
					>
						{choice}
					</option>
				))}
			</select>
		</div>
			)
}

function DatalistFilter({
	name,
	label,
	value,
	choices,
	handleFilterChange,
}) {
	let id = `id_${name}`
	let choicesId = `${name}_choices`

	return (
		<div className={"filters__field-row"}>
			<div className="filters__field-label">
				<label htmlFor={id}>{label}</label>
			</div>
			<input
				value={value}
				type="text"
				name={name}
				className="text-field--single"
				autoComplete="off"
				onChange={handleFilterChange}
				id={id}
				list={choicesId}
			/>

			<datalist
				id={choicesId}
			>
				{choices.map((choice, index) => (
					<option key={index} value={choice}></option>
				))}
			</datalist>
		</div>
	)
}

function RadioFilter({
	name,
	label,
	value,
	options,
	handleFilterChange,
}) {
	let id = `id_${name}`

	return (
		<div className={"filters__field-row"}>
			<div className="filters__field-label">
				{label}
			</div>

			<ul id={id}>
				{options.map( (option, index) => {
					let optionId = `id_${name}_${index}`
					return (
						<li key={option.value}>
							<label htmlFor={optionId}>
								<input
									type="radio"
									name={name}
									value={option.value}
									checked={value === option.value}
									id={optionId}
									onChange={handleFilterChange}
								/>
								{option.label}
							</label>
						</li>
					)
				})}
			</ul>
		</div>
	)
}

RadioFilter.defaultProps = {
	options: [
		{ label: 'Unknown', value: 'NOTHING' },
		{ label: 'Yes', value: 'JUST_TRUE' },
		{ label: 'No', value: 'JUST_FALSE' },
	],
}


export function BoolFilter(props) {
	return (
		<RadioFilter
			options={[
				{ label: 'Yes', value: '1' },
				{ label: 'No', value: '0' },
			]}
			{...props}
		/>
	)
}

export default function FilterSet({ filters, handleFilterChange, filterParameters, width, dataset, filterWithout}) {
	const components = filters.map((filter, index) => {
		if (filter.name === 'search') {
			return
		} else if (filter.name == 'tags') {
			return (
				<details
					className="filters__group filters__form--category"
					open={Boolean(filterParameters.tags.parameters)}
					key={index}
				>
					<summary className="filters__form-summary">
						<h3 className="filter__heading">Tag</h3>
					</summary>
					<TagFilter
						width={width}
						dataset={dataset}
						filterParameters={filterParameters.tags.parameters}
					/>
				</details>
			)
		}
		if (filter.type === 'text') {
			return (
				<TextFilter
					key={index}
					name={filter.name}
					label={filter.title}
					value={filterParameters[filter.name].parameters || ""}
					handleFilterChange={handleFilterChange}
				/>
			)
		} else if (filter.type === 'date') {
			return (
				<DateFilter
					key={index}
					name={filter.name}
					label={filter.title}
					value={filterParameters[filter.name].parameters || {}}
				/>
			)
		} else if (filter.type === 'radio') {
			return (
				<RadioFilter
					key={index}
					name={filter.name}
					label={filter.title}
					value={filterParameters[filter.name].parameters || ""}
					handleFilterChange={handleFilterChange}
				/>
			)
		} else if (filter.type === 'bool') {
			return (
				<BoolFilter
					key={index}
					name={filter.name}
					label={filter.title}
					value={filterParameters[filter.name].parameters || ""}
					handleFilterChange={handleFilterChange}
				/>
			)
		} else if (filter.type === 'choice') {
			return (
				<SelectFilter
					key={index}
					name={filter.name}
					label={filter.title}
					choices={filter.choices}
					value={filterParameters[filter.name].parameters || ""}
					handleFilterChange={handleFilterChange}
				/>
			)
		} else if (filter.type == 'autocomplete') {
			return (
				<DatalistFilter
					key={index}
					name={filter.name}
					label={filter.title}
					choices={filter.choices}
					value={filterParameters[filter.name].parameters || ""}
					handleFilterChange={handleFilterChange}
				/>
			)
		} else {
			console.warn(`no filter defined for type "${filter.type}"`)
		}
	})
	return components
}



FilterSet.propTypes = {
	filters: PropTypes.array.isRequired,
	handleFilterChange: PropTypes.func.isRequired,
	filterParameters: PropTypes.object.isRequired,
}
