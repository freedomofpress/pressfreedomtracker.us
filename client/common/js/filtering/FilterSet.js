import React, { PropTypes } from 'react'
import {
	AutocompleteInput,
	BoolInput,
	ChoiceInput,
	DateRangeInput,
	IntInput,
	TextInput,
	RadioPillInput,
} from '~/filtering/Inputs'

function FilterSet({ filters, handleFilterChange, filterValues }) {
	const components = filters.map((filter, index) => {
		if (filter.type === 'choice') {
			return (
				<ChoiceInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={filter.title}
					filter={filter.name}
					choices={filter.choices}
					key={index}
				/>
			)
		} else if (filter.type === 'date') {
			return (
				<DateRangeInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={filter.title}
					filter={filter.name}
					key={index}
				/>
			)
		} else if (filter.type === 'text') {
			return (
				<TextInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={filter.title}
					filter={filter.name}
					key={index}
				/>
			)
		} else if (filter.type === 'radio') {
			return (
				<RadioPillInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={filter.title}
					filter={filter.name}
					key={index}
				/>
			)
		} else if (filter.type === 'bool') {
			return (
				<BoolInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={`${filter.title}`}
					filter={filter.name}
					key={index}
				/>
			)
		} else if (filter.type === 'int') {
			return (
				<IntInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={`${filter.title}`}
					filter={filter.name}
					key={index}
					units={filter.units}
				/>
			)
		} else if (filter.type === 'autocomplete') {
			return (
				<AutocompleteInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={filter.title}
					filter={filter.name}
					type={filter.autocomplete_type}
					isSingle={!filter.many}
					key={index}
				/>
			)
		}
		console.error('Unknown filter type.')
		return null
	})
	return (
		<div className="filters__set">
			{components}
		</div>
	)
}


FilterSet.propTypes = {
	filters: PropTypes.array.isRequired,
	handleFilterChange: PropTypes.func.isRequired,
	filterValues: PropTypes.object.isRequired,
}


export default FilterSet
