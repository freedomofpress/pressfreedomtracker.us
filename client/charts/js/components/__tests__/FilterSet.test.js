import React from 'react'
import Renderer from 'react-test-renderer'
import FilterSet from '../FilterSet'

test('renders FilterSet with mocked data', () => {
	expect(Renderer.create(
		<FilterSet
			filters={[
				{
					"title": "Search terms",
					"type": "text",
					"name": "search"
				},
				{
					"title": "Took place",
					"type": "date",
					"name": "date"
				},
				{
					"title": "City",
					"type": "text",
					"name": "city"
				},
				{
					"title": "State",
					"type": "autocomplete",
					"name": "state",
					"autocomplete_type": "incident.State",
					"choices": [
						"Alaska",
						"Alabama",
						"Arkansas",
						"American Samoa",
						"Arizona",
						"California",
						"Colorado",
						"Connecticut",
						"District of Columbia",
						"Delaware",
						"Florida",
						"Georgia",
						"Guam",
						"Hawaii",
						"Iowa",
						"Idaho",
						"Illinois",
						"Indiana",
						"Kansas"
					],
					"many": false
				},
				{
					"title": "Targeted any of these journalists",
					"type": "autocomplete",
					"name": "targeted_journalists",
					"autocomplete_type": "incident.Journalist",
					"choices": [
						"Abigail Peterson",
						"Adam Buchanan",
						"Adam Lopez MD",
						"Alicia Savage",
						"Amanda Rosales",
						"Angela Davis",
						"Angela Johnson",
						"Anthony Black",
						"Anthony Ellison",
						"Anthony Poole",
						"April Olson",
						"Ariel Hill",
						"Austin Henry"
					],
					"many": true
				},
				{
					"title": "Targeted Institutions",
					"type": "autocomplete",
					"name": "targeted_institutions",
					"autocomplete_type": "incident.Institution",
					"choices": [
						"The Aaronmouth Post 141",
						"The Alexanderbury Sun 191",
						"The Alexisberg Post 155",
						"The Amberport Tribune 100",
						"The Andrewview Tribune 39",
						"The Annaberg Tribune 145",
						"The Anneside Post 41",
						"The Ashleymouth Tribune 184",
						"The Austinborough Sun 164",
						"The Banksberg Herald 142",
						"The Batesland Post 126",
						"The Beckyview Daily News 128",
						"The Bellburgh Post 135",
						"The Benderton Sun 102",
						"The Bethfort Daily News 7",
						"The Blevinsland Tribune 200",
						"The Bobbymouth Tribune 170",
						"The Bradleyshire Herald 198",
						"The Brittanyshire Sun 74",
						"The Brownview Daily News 148",
						"The Bryantmouth Daily News 11"
					],
					"many": true
				},
				{
					"title": "Has any of these tags",
					"type": "autocomplete",
					"name": "tags",
					"autocomplete_type": "common.CommonTag",
					"choices": [
						"ammonites",
						"ants",
						"aphids",
						"baboon",
						"bamboo",
						"bees",
						"centipedes",
						"chrysanthemum",
						"civet",
						"clams",
						"copepods",
						"crabs",
						"crickets",
						"crocodiles",
						"cuttlefish",
						"doves",
						"dragonflies"
					],
					"many": true
				},
				{
					"title": "Case number",
					"type": "text",
					"name": "case_number"
				},
				{
					"title": "Type of case",
					"type": "choice",
					"name": "case_type",
					"choices": [
						[
							"CIVIL",
							"Civil"
						],
						[
							"CLASS_ACTION",
							"Class Action"
						]
					]
				},
				{
					"title": "Legal case statuses",
					"type": "choice",
					"name": "case_statuses",
					"choices": [
						[
							"ONGOING",
							"ongoing"
						],
						[
							"SETTLED",
							"settled"
						],
						[
							"DISMISSED",
							"dismissed"
						],
						[
							"UNKNOWN",
							"unknown"
						],
						[
							"APPEALED",
							"appealed"
						]
					]
				}
			]}
			width={500}
			handleFilterChange={() => {}}
			filterParameters={{
				"filterTimeMonths": {
					"type": "timeMonths",
					"parameters": {
						"min": "2022-03-20T00:00:00.000Z",
						"max": "2023-02-18T00:00:00.000Z"
					},
					"enabled": true
				},
				"filterTimeYears": {
					"type": "timeYears",
					"parameters": [
						2022,
						2023
					],
					"enabled": false
				},
				"filterState": {
					"type": "state",
					"parameters": "All",
					"enabled": true
				},
				"date": {
					"enabled": false,
					"type": "date",
					"parameters": {
						"min": null,
						"max": null
					}
				},
				"detention_date": {
					"enabled": false,
					"type": "date",
					"parameters": {
						"min": null,
						"max": null
					}
				},
				"release_date": {
					"enabled": false,
					"type": "date",
					"parameters": {
						"min": null,
						"max": null
					}
				},
				"search": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"recently_updated": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"city": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"state": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"targeted_journalists": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"targeted_institutions": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"lawsuit_name": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"venue": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"case_statuses": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"case_type": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"case_number": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"arrest_status": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"arresting_authority": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"status_of_charges": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"charges": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"unnecessary_use_of_force": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"border_point": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"stopped_previously": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"target_us_citizenship_status": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"denial_of_entry": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"target_nationality": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"did_authorities_ask_for_device_access": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"did_authorities_ask_for_social_media_user": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"did_authorities_ask_for_social_media_pass": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"did_authorities_ask_about_work": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"were_devices_searched_or_seized": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"politicians_or_public_figures_involved": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"equipment_seized": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"status_of_seized_equipment": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"is_search_warrant_obtained": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"actor": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"assailant": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"was_journalist_targeted": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"charged_under_espionage_act": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"subpoena_type": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"subpoena_statuses": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"held_in_contempt": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"detention_status": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"third_party_business": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"legal_order_type": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"equipment_broken": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"status_of_prior_restraint": {
					"enabled": false,
					"type": "string",
					"parameters": null
				},
				"tags": {
					"enabled": true,
					"type": "stringset",
					"parameters": {}
				},
				"categories": {
					"enabled": true,
					"type": "stringset",
					"parameters": {}
				}
			}}
			dataset={[
				{
					"date": "2023-02-12T00:00:00.000Z",
					"city": "Thomasland",
					"state": "ID",
					"latitude": null,
					"longitude": null,
					"categories": [
						"Arrest / Criminal Charge"
					],
					"tags": [
						"lemur"
					]
				},
				{
					"date": "2022-10-23T00:00:00.000Z",
					"city": "New Brooke",
					"state": "NC",
					"latitude": null,
					"longitude": null,
					"categories": [
						"Arrest / Criminal Charge"
					],
					"tags": [
						"baboon"
					]
				},
				{
					"date": "2022-05-12T00:00:00.000Z",
					"city": "East Mary",
					"state": "CT",
					"latitude": null,
					"longitude": null,
					"categories": [
						"Equipment Search or Seizure"
					],
					"tags": [
						"civet"
					]
				},
				{
					"date": "2022-07-24T00:00:00.000Z",
					"city": "Josephmouth",
					"state": "AL",
					"latitude": null,
					"longitude": null,
					"categories": [
						"Equipment Search or Seizure"
					],
					"tags": [
						"hyena"
					]
				},
			]}
		/>
	)).toMatchSnapshot();
});
