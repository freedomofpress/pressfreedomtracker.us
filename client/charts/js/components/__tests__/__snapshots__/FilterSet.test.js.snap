// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FilterSet with mocked data 1`] = `
[
  <div
    className="filters__field-row"
  >
    <div
      className="filters__field-label"
    >
      <label
        htmlFor="id_date_after"
      >
        Took place
         after
      </label>
    </div>
    <input
      id="id_date_after"
      name="date"
      onChange={[Function]}
      type="date"
      value=""
    />
  </div>,
  <div
    className="filters__field-row"
  >
    <div
      className="filters__field-label"
    >
      <label
        htmlFor="id_date_before"
      >
        Took place
         before
      </label>
    </div>
    <input
      id="id_date_before"
      name="date"
      onChange={[Function]}
      type="date"
      value=""
    />
  </div>,
  <div
    className="filters__field-row"
  >
    <div
      className="filters__field-label"
    >
      <label
        htmlFor="id_city"
      >
        City
      </label>
    </div>
    <input
      autoComplete="off"
      className="text-field--single"
      id="id_city"
      name="city"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>,
  <div
    className="filters__field-row"
  >
    <div
      className="filters__field-label"
    >
      <label
        htmlFor="id_state"
      >
        State
      </label>
    </div>
    <input
      autoComplete="off"
      className="text-field--single"
      id="id_state"
      list="state_choices"
      name="state"
      onChange={[Function]}
      type="text"
      value=""
    />
    <datalist
      id="state_choices"
    >
      <option
        value="Alaska"
      />
      <option
        value="Alabama"
      />
      <option
        value="Arkansas"
      />
      <option
        value="American Samoa"
      />
      <option
        value="Arizona"
      />
      <option
        value="California"
      />
      <option
        value="Colorado"
      />
      <option
        value="Connecticut"
      />
      <option
        value="District of Columbia"
      />
      <option
        value="Delaware"
      />
      <option
        value="Florida"
      />
      <option
        value="Georgia"
      />
      <option
        value="Guam"
      />
      <option
        value="Hawaii"
      />
      <option
        value="Iowa"
      />
      <option
        value="Idaho"
      />
      <option
        value="Illinois"
      />
      <option
        value="Indiana"
      />
      <option
        value="Kansas"
      />
    </datalist>
  </div>,
  <div
    className="filters__field-row"
  >
    <div
      className="filters__field-label"
    >
      <label
        htmlFor="id_targeted_journalists"
      >
        Targeted any of these journalists
      </label>
    </div>
    <input
      autoComplete="off"
      className="text-field--single"
      id="id_targeted_journalists"
      list="targeted_journalists_choices"
      name="targeted_journalists"
      onChange={[Function]}
      type="text"
      value=""
    />
    <datalist
      id="targeted_journalists_choices"
    >
      <option
        value="Abigail Peterson"
      />
      <option
        value="Adam Buchanan"
      />
      <option
        value="Adam Lopez MD"
      />
      <option
        value="Alicia Savage"
      />
      <option
        value="Amanda Rosales"
      />
      <option
        value="Angela Davis"
      />
      <option
        value="Angela Johnson"
      />
      <option
        value="Anthony Black"
      />
      <option
        value="Anthony Ellison"
      />
      <option
        value="Anthony Poole"
      />
      <option
        value="April Olson"
      />
      <option
        value="Ariel Hill"
      />
      <option
        value="Austin Henry"
      />
    </datalist>
  </div>,
  <div
    className="filters__field-row"
  >
    <div
      className="filters__field-label"
    >
      <label
        htmlFor="id_targeted_institutions"
      >
        Targeted Institutions
      </label>
    </div>
    <input
      autoComplete="off"
      className="text-field--single"
      id="id_targeted_institutions"
      list="targeted_institutions_choices"
      name="targeted_institutions"
      onChange={[Function]}
      type="text"
      value=""
    />
    <datalist
      id="targeted_institutions_choices"
    >
      <option
        value="The Aaronmouth Post 141"
      />
      <option
        value="The Alexanderbury Sun 191"
      />
      <option
        value="The Alexisberg Post 155"
      />
      <option
        value="The Amberport Tribune 100"
      />
      <option
        value="The Andrewview Tribune 39"
      />
      <option
        value="The Annaberg Tribune 145"
      />
      <option
        value="The Anneside Post 41"
      />
      <option
        value="The Ashleymouth Tribune 184"
      />
      <option
        value="The Austinborough Sun 164"
      />
      <option
        value="The Banksberg Herald 142"
      />
      <option
        value="The Batesland Post 126"
      />
      <option
        value="The Beckyview Daily News 128"
      />
      <option
        value="The Bellburgh Post 135"
      />
      <option
        value="The Benderton Sun 102"
      />
      <option
        value="The Bethfort Daily News 7"
      />
      <option
        value="The Blevinsland Tribune 200"
      />
      <option
        value="The Bobbymouth Tribune 170"
      />
      <option
        value="The Bradleyshire Herald 198"
      />
      <option
        value="The Brittanyshire Sun 74"
      />
      <option
        value="The Brownview Daily News 148"
      />
      <option
        value="The Bryantmouth Daily News 11"
      />
    </datalist>
  </div>,
  <div
    className="filters__field-row"
  >
    <div
      className="filters__field-label"
    >
      <label
        htmlFor="id_case_number"
      >
        Case number
      </label>
    </div>
    <input
      autoComplete="off"
      className="text-field--single"
      id="id_case_number"
      name="case_number"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>,
  <div
    className="filters__field-row"
  >
    <div
      className="filters__field-label"
    >
      <label
        htmlFor="id_case_type"
      >
        Type of case
      </label>
    </div>
    <select
      id="id_case_type"
      name="case_type"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        ------
      </option>
      <option
        value="CIVIL"
      >
        Civil
      </option>
      <option
        value="CLASS_ACTION"
      >
        Class Action
      </option>
    </select>
  </div>,
  <div
    className="filters__field-row"
  >
    <div
      className="filters__field-label"
    >
      <label
        htmlFor="id_case_statuses"
      >
        Legal case statuses
      </label>
    </div>
    <select
      id="id_case_statuses"
      name="case_statuses"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        ------
      </option>
      <option
        value="ONGOING"
      >
        ongoing
      </option>
      <option
        value="SETTLED"
      >
        settled
      </option>
      <option
        value="DISMISSED"
      >
        dismissed
      </option>
      <option
        value="UNKNOWN"
      >
        unknown
      </option>
      <option
        value="APPEALED"
      >
        appealed
      </option>
    </select>
  </div>,
]
`;
